test/test_autograd.py

```diff
@@ -76,6 +76,11 @@ def graph_desc(fn):
 
 class TestAutograd(TestCase):
 
+    def setUp(self):
+        super(TestAutograd, self).setUp()
+        force_parallel = bool(hash(self.id()) % 2)
```

Comment:
We've been working to reduce non-determinism in tests. We shouldn't explicitly add non-determinism.

Reply:
FWIW both paths should give the same results, but I agree that it might be a bad idea due to Python seeding its hash function differently at run time. I'll use a deterministic hash function.